File: app/src/test/java/com/cargopull/executor_driver/presentation/balance/BalanceViewStatePendingTest.java
package com.cargopull.executor_driver.presentation.balance;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotEquals;
import static org.mockito.Mockito.only;
import static org.mockito.Mockito.verify;
import static org.mockito.Mockito.verifyNoMoreInteractions;

import com.cargopull.executor_driver.presentation.ViewState;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.junit.MockitoJUnitRunner;

@RunWith(MockitoJUnitRunner.class)
public class BalanceViewStatePendingTest {

  private BalanceViewStatePending viewState;

  @Mock
  private BalanceViewActions viewActions;
  @Mock
  private ViewState<BalanceViewActions> parentViewState;
  @Mock
  private ViewState<BalanceViewActions> parentViewState1;

  @Test
  public void testActions() {
    // Дано:
    viewState = new BalanceViewStatePending(parentViewState);

    // Действие:
    viewState.apply(viewActions);

    // Результат:
    verify(viewActions).showBalancePending(true);
    verify(parentViewState, only()).apply(viewActions);
    verifyNoMoreInteractions(viewActions);
  }

  @Test
  public void testActionsWithNull() {
    // Дано:
    viewState = new BalanceViewStatePending(null);

    // Действие:
    viewState.apply(viewActions);

    // Результат:
    verify(viewActions).showBalancePending(true);
    verifyNoMoreInteractions(viewActions);
  }

  @Test
  public void testEquals() {
    viewState = new BalanceViewStatePending(parentViewState);
    assertEquals(viewState, new BalanceViewStatePending(parentViewState));
    assertNotEquals(viewState, new BalanceViewStatePending(parentViewState1));
    assertNotEquals(viewState, new BalanceViewStatePending(null));
  }
}

File: app/src/test/java/com/cargopull/executor_driver/presentation/balance/BalanceViewStatePendingTest.kt
package com.cargopull.executor_driver.presentation.balance

import com.cargopull.executor_driver.presentation.ViewState
import org.junit.Assert.assertEquals
import org.junit.Assert.assertNotEquals
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.Mock
import org.mockito.Mockito.*
import org.mockito.junit.MockitoJUnitRunner

@RunWith(MockitoJUnitRunner::class)
class BalanceViewStatePendingTest {

    private lateinit var viewState: BalanceViewStatePending

    @Mock
    private lateinit var viewActions: BalanceViewActions
    @Mock
    private lateinit var parentViewState: ViewState<BalanceViewActions>
    @Mock
    private lateinit var parentViewState1: ViewState<BalanceViewActions>

    @Test
    fun testActions() {
        // Given:
        viewState = BalanceViewStatePending(parentViewState)

        // Action:
        viewState.apply(viewActions)

        // Effect:
        //    verify(viewActions).showBalancePending(true);
        verify<ViewState<BalanceViewActions>>(parentViewState, only()).apply(viewActions)
        verify(viewActions).blockWithPending("BalanceViewState")
        verifyNoMoreInteractions(viewActions)
    }

    @Test
    fun testActionsWithNull() {
        // Given:
        viewState = BalanceViewStatePending(null)

        // Action:
        viewState.apply(viewActions)

        // Effect:
        verify(viewActions).blockWithPending("BalanceViewState")
        verifyNoMoreInteractions(viewActions)
    }

    @Test
    fun testEquals() {
        viewState = BalanceViewStatePending(parentViewState)
        assertEquals(viewState, BalanceViewStatePending(parentViewState))
        assertNotEquals(viewState, BalanceViewStatePending(parentViewState1))
        assertNotEquals(viewState, BalanceViewStatePending(null))
    }
}